pkg/validation/options.go

```diff
@@ -29,28 +27,6 @@ func Validate(o *options.Options) error {
 	msgs = configureLogger(o.Logging, msgs)
 	msgs = parseSignatureKey(o, msgs)
 
-	if o.SSLInsecureSkipVerify {
-		// InsecureSkipVerify is a configurable option we allow
-		/* #nosec G402 */
-		insecureTransport := &http.Transport{
-			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
-		}
-		http.DefaultClient = &http.Client{Transport: insecureTransport}
-	} else if len(o.Providers[0].CAFiles) > 0 {
-		pool, err := util.GetCertPool(o.Providers[0].CAFiles)
-		if err == nil {
-			transport := http.DefaultTransport.(*http.Transport).Clone()
-			transport.TLSClientConfig = &tls.Config{
-				RootCAs:    pool,
-				MinVersion: tls.VersionTLS12,
-			}
-
-			http.DefaultClient = &http.Client{Transport: transport}
-		} else {
-			msgs = append(msgs, fmt.Sprintf("unable to load provider CA file(s): %v", err))
-		}
-	}
-
```

Comment:
@tuunit So I've been looking through the code for another potential request, and this block from the main branch jumped out at me. In my environments, I have to push outbound requests through an HTTP proxy and having `http.DefaultTransport` makes sure requests to the provider will honor those vars.

Correct me if I'm wrong, but if users with similar env requirements set `--ssl-insecure-skip-verify`, then oauth2-proxy wouldn't use that proxy for outbound requests to the provider? I personally have yet to see this because the Keycloak I use is within the NO_PROXY range, but I do have upcoming deployments where that may not be the case.

If we think that's correct and setting that flag would block pushing requests through the proxy, I'm happy to put up a separate PR against main to use the DefaultTransport.

Reply:
Thnaks @ethanchowell for your time and keeping this PR up to date! Unfortunately, I was rather busy and sick lately. Therefore I wasn't able to check it out again. I try my best to get around to it soon.